src/server/pkg/obj/microsoft_client.go

```diff
@@ -135,103 +114,69 @@ func (c *microsoftClient) IsIgnorable(err error) bool {
 	span, _ := tracing.AddSpanToAnyExisting(w.ctx, "/microsoftWriter/Write")
 	defer tracing.FinishAnySpan(span)
 	if w.err != nil {
-		// w.err wasn't generated by putting this block but a previous block,
-		// we return it here though because PutBlock is called async so we
-		// can't return it from the correct call but we also don't want to wait
-		// until the end to handle errors.
 		return 0, w.err
 	}
-	nBytes := 0
-	for {
-		if w.buf.Len()+len(b) >= maxBlockSize {
-			offset := maxBlockSize - w.buf.Len()
-			w.buf.Write(b[:offset])
-			w.writeBlock(w.buf.Bytes())
-			nBytes += offset
-			w.buf = bytes.NewBuffer(bufPool.GetBuffer()[:0])
-			b = b[offset:]
-		} else {
-			w.buf.Write(b)
-			nBytes += len(b)
-			break
-		}
-	}
-	return nBytes, nil
-	// TODO according to msft docs a blob can have at most 100,000 uncommitted
-	// blocks and at most 200,000 MB so this code will probably break with
-	// objects over 200GB, right now that's a rare case. But when we do hit
-	// this we should add a check in this function that commits the blocks with
-	// a PutBlockList call when there's more than 200,000 MB outstanding.
-}
-
-func blockID(n int) string {
-	// according to msft docs: Block IDs are strings of equal length within a
-	// blob. Block client code usually uses base-64 encoding to normalize
-	// strings into equal lengths. When using base-64 encoding, the pre-encoded
-	// string must be 64 bytes or less. Block ID values can be duplicated in
-	// different blobs.
-	return base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("%011d\n", n)))
+	return w.w.Write(data)
 }
 
-func (w *microsoftWriter) writeBlock(b []byte) {
-	block := w.nBlocks
-	w.nBlocks++
+func (w *microsoftWriter) writeBlock(block []byte) error {
+	blockID := blockID(w.numBlocks)
+	w.numBlocks++
 	w.limiter.Acquire()
 	w.eg.Go(func() error {
 		defer w.limiter.Release()
-		defer bufPool.PutBuffer(b)
-		if err := w.blobClient.PutBlock(w.container, w.blob, blockID(block), b); err != nil {
+		defer bufPool.Put(block[:cap(block)])
+		if err := w.blob.PutBlock(blockID, block, nil); err != nil {
 			w.err = err
 			return err
 		}
 		return nil
 	})
+	return nil
+}
+
+func blockID(n int) string {
+	return base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("%011d\n", n)))
 }
 
 func (w *microsoftWriter) Close() error {
 	span, _ := tracing.AddSpanToAnyExisting(w.ctx, "/microsoftWriter/Close")
 	defer tracing.FinishAnySpan(span)
-	if w.buf.Len() > 0 {
-		w.writeBlock(w.buf.Bytes())
+	if err := w.w.Close(); err != nil {
+		return err
 	}
 	if err := w.eg.Wait(); err != nil {
 		return err
 	}
-	blocks := make([]storage.Block, w.nBlocks)
+	// Finalize the blocks.
+	blocks := make([]storage.Block, w.numBlocks)
 	for i := range blocks {
 		blocks[i] = storage.Block{ID: blockID(i), Status: storage.BlockStatusUncommitted}
 	}
-	return w.blobClient.PutBlockList(w.container, w.blob, blocks)
+	return w.blob.PutBlockList(blocks, nil)
```

Comment:
I was hoping that the new azure-sdk would have support for contexts, but it looks like that's not the case. I just want to confirm to make sure I'm reading this right and that is indeed the case. It seems pretty weird they wouldn't have adopted context yet.

Reply:
Yeah, no support for contexts.